### PR TITLE
Feature/hocs 2659 message generator

### PR DESCRIPTION
**Pull request by philbarton**
This is the first attempt at the k8s config for the message generator. 

The config is for a kubernetes 'job'. The intention is to run the job is to allow testers to run the generator from drone.

The parameters 
RUN_CONFIG_NUM_MESSAGES ( can be blank, defaults to 1)
and 
RUN_CONFIG_COMPLAINT_TYPE ( can be blank, so selects a random message type)
will be passed in through the drone1 promote command.

I don't know how to test this without deploying it, so this might be an iterative process.

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+set -euo pipefail
+
+export KUBE_NAMESPACE=${ENVIRONMENT}
+export KUBE_SERVER=${KUBE_SERVER}
```
Comment (zuzak) [inline deploy.sh:5]: As this is notprod only you may as well specify the KUBE_SERVER here.

File: kd/message-generator-job.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+metadata:
+  name: hocs-message-generator
+  labels:
+    version: {{.VERSION}}
+spec:
+  template:
+    metadata:
+      labels:
+        name: hocs-message-generator
+        role: hocs-backend
```
Comment (zuzak) [inline kd/message-generator-job.yaml:13]: The `hocs-backend` role is mostly (only?) used to specify network policies. If this job doesn't need to interact with any other pods, maybe remove this line?

File: kd/message-generator-job.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                drop:
+                  - SETUID
+                  - SETGID
+            envFrom:
+              - configMapRef:
+                  name: hocs-queue-config
+            env:
+              - name: JAVA_OPTS
+                value: >
+                  -Xms768m -Xmx4096m -XX:+UseG1GC -Djavax.net.ssl.trustStore=/etc/keystore/truststore.jks
```
Comment (zuzak) [inline kd/message-generator-job.yaml:32]: You're hardcoding the JVM memory stuff but you're not specifying a `resources` block in Kubernetes. Although the defaults in ACP are sensible, it's probably worth figuring out a proper value.

**Review by zuzak — APPROVED** [review]
Need to do that bash variable fix, but otherwise fine

File: kd/message-generator-job.yaml (hunk 3 above)
Comment (philbarton) [reply kd/message-generator-job.yaml]: I've set the resources. JDK 11 should respect the k8s setting, so removed explicit settings.

File: kd/message-generator-job.yaml (hunk 2 above)
Comment (philbarton) [reply kd/message-generator-job.yaml]: removed

File: deploy.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+export KUBE_TOKEN=${KUBE_TOKEN}
+export VERSION=${VERSION}
+
+export RUN_CONFIG_NUM_MESSAGES=${RUN_CONFIG_NUM_MESSAGES=1}
```
[the remark this answers is not in the document]
Comment (philbarton) [reply deploy.sh]: Odd, it worked in scripts from previous project! I've changed it as suggested.

File: deploy.sh (hunk 1 above)
Comment (philbarton) [reply deploy.sh]: Done

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+.idea
+*.iml
```
[the remark this answers is not in the document]
Comment (philbarton) [reply .gitignore:2]: I agree, and a shared IntelliJ config. Maybe not for this merge....
I'll add an IDEAS ticket.

**Review by zuzak — APPROVED** [review]

**Pull request merged**